src/s3fs.cpp

```diff
@@ -3078,28 +3078,17 @@ static int s3fs_check_service(void)
   }
 
   S3fsCurl s3fscurl;
-  if(-1 == s3fscurl.CheckBucket()){
+  if(1 == s3fscurl.CheckBucket()){
     fprintf(stderr, "%s: Failed to access bucket.\n", program_name.c_str());
     return EXIT_FAILURE;
   }
   long responseCode = s3fscurl.GetLastResponseCode();
 
-  if(responseCode == 400){
-    if(!S3fsCurl::IsSignatureV4()){
-      // signature version 2
-      fprintf(stderr, "%s: Bad Request\n", program_name.c_str());
-      return EXIT_FAILURE;
-    }
-    if(is_specified_endpoint){
-      // if specifies endpoint, do not retry to connect.
-      fprintf(stderr, "%s: Bad Request\n", program_name.c_str());
-      return EXIT_FAILURE;
-    }
-
+  if(responseCode == 400) {
     // check region error for signature version 4
     BodyData* body = s3fscurl.GetBodyData();
     string    expectregion;
-    if(check_region_error(body->str(), expectregion)){
+    if(check_region_error(body->str(), expectregion) && !is_specified_endpoint){
```

Comment:
Checking "!is_specified_endpoint" should be first. because the checking cost is is_specified_endpoint < check_region_error flag.